prow/integ-suite-ocp.sh

```diff
@@ -69,21 +77,49 @@ HUB="image-registry.openshift-image-registry.svc:5000/${NAMESPACE}"
 # Build the base command and store it in a variable.
 # TODO: execute the test by running make target. Do we need first to add a skip flag to the make target to be able to skip failing test on OCP
 # All the flags are needed to run the integration tests on OCP
+# Initialize base_cmd
+base_cmd=""
+
+# Set up test command and parameters
+setup_junit_report() {
+    export ISTIO_BIN="${GOPATH}/bin"
+    echo "ISTIO_BIN: ${ISTIO_BIN}"
+
+    JUNIT_REPORT=$(which go-junit-report 2>/dev/null)
+    if [ -z "$JUNIT_REPORT" ]; then
+        JUNIT_REPORT="${ISTIO_BIN}/go-junit-report"
+    fi
+    echo "JUNIT_REPORT: ${JUNIT_REPORT}"
+}
+
+# Base command setup
 base_cmd="go test -p 1 -v -count=1 -tags=integ -vet=off -timeout 60m ./tests/integration/${TEST_SUITE}/... \
 --istio.test.ci \
 --istio.test.pullpolicy=IfNotPresent \
---istio.test.work_dir=result \
+--istio.test.work_dir=${ARTIFACTS_DIR} \
 --istio.test.skipTProxy=true \
 --istio.test.skipVM=true \
---istio.test.kube.helm.values=profile=openshift,global.platform=openshift \
+--istio.test.kube.helm.values='profile=openshift,global.platform=openshift' \
 --istio.test.istio.enableCNI=true \
 --istio.test.hub=\"${HUB}\" \
 --istio.test.tag=\"${TAG}\""
 
-# Check if SKIP_TESTS is non-empty and append the -skip flag if it is.
+# Check if TEST_SUITE is 'helm' and add specific flag
+if [ "${TEST_SUITE}" = "helm" ]; then
+    base_cmd+=" --istio.test.openshift"
+fi
+
+# Append skip tests flag if SKIP_TESTS is set
 if [ -n "${SKIP_TESTS}" ]; then
-  base_cmd+=" -skip '${SKIP_TESTS}'"
+    base_cmd+=" -skip '${SKIP_TESTS}'"
+fi
+
+# Setup JUnit report and modify base command if running in Prow mode
+if [ "${PROW}" == "true" ]; then
```

Comment:
> Any reason to only run junit when in prow? Is it much slow?

No, is because locally you will need to have the junit report package locally to be able to run. We can change the variable name to something more clear like: `TEST_OUTPUT_FORMAT` default is JUNIT and add a comment that you will need to have the package (Upstream does something similar)

> Another question: Do we need junit for anything at all?
The main reason is to be able to track easily the test cases executed. This can be added later to a test grid to have more analysis and I think is very useful to check: 
* To check the test coverage
* When multiple test case fails, it is useful to see the total amount of test cases failing. Specially when the log is chaos because of the failures

Reply:
And Downstream we can export those junit files to a report tool to track the results over the time

Reply:
Note that the builder image already comes with junit. There's no need to install in on the host if we build within the container.

Reply:
Yes, I have another task open because running from the make target in a container fails against ocp cluster. I need to investigate that

BTW this si something that we need to aim to have the results as upstream have it: https://testgrid.k8s.io/istio_istio#integ-helm_istio